internal/apis/certmanager/validation/certificate.go

```diff
@@ -47,12 +48,56 @@ func ValidateCertificateSpec(crt *internalcmapi.CertificateSpec, fldPath *field.
 
 	el = append(el, validateIssuerRef(crt.IssuerRef, fldPath)...)
 
-	if len(crt.CommonName) == 0 && len(crt.DNSNames) == 0 && len(crt.URISANs) == 0 && len(crt.EmailSANs) == 0 && len(crt.IPAddresses) == 0 {
+	var commonName = crt.CommonName
+	if crt.LiteralSubject != "" {
+
+		if !utilfeature.DefaultFeatureGate.Enabled(feature.LiteralCertificateSubject) {
+			el = append(el, field.Forbidden(fldPath.Child("literalSubject"), "Feature gate LiteralCertificateSubject must be enabled on both webhook and controller to use the alpha `literalSubject` field"))
+		}
+
+		sequence, err := pki.ParseSubjectStringToRdnSequence(crt.LiteralSubject)
+		if err != nil {
+			el = append(el, field.Invalid(fldPath.Child("literalSubject"), crt.LiteralSubject, err.Error()))
+		}
+
+		// Must contain a CN
+		for _, rdns := range sequence {
+			for _, atv := range rdns {
+				if atv.Type.Equal(pki.OIDConstants.CommonName) {
+					if str, ok := atv.Value.(string); ok {
+						commonName = str
+					} else {
+						el = append(el, field.Invalid(fldPath.Child("literalSubject"), atv.Value, "Field with type CN should be a string"))
+					}
+				}
+			}
+		}
+
+		// Should not contain unrecognized OIDs
+		for _, rdns := range sequence {
+			for _, atv := range rdns {
+				if atv.Type.Equal(nil) {
+					el = append(el, field.Invalid(fldPath.Child("literalSubject"), crt.LiteralSubject, fmt.Sprintf("Literal subject contains unrecognized key with value [%s]", atv.Value)))
+				}
+			}
+		}
+
+		if len(crt.CommonName) != 0 {
+			el = append(el, field.Invalid(fldPath.Child("commonName"), crt.CommonName, "When providing a `LiteralSubject` no `commonName` may be provided."))
+		}
+
+		if crt.Subject != nil && len(crt.Subject.Organizations)+len(crt.Subject.Countries)+len(crt.Subject.OrganizationalUnits)+len(crt.Subject.Localities)+len(crt.Subject.Provinces)+len(crt.Subject.StreetAddresses)+len(crt.Subject.PostalCodes) != 0 {
+			el = append(el, field.Invalid(fldPath.Child("subject"), crt.Subject, "When providing a `LiteralSubject` no `Subject` properties may be provided with the exception of `Subject.serialNumber`"))
+		}
```

Comment:
note: I was incorrect in advising to treat serialNumber as a special case. See my comment [here](https://github.com/cert-manager/cert-manager/pull/5002#issuecomment-1088567274).

Adding this comment for anyone looking back in the future!